android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.java

```diff
@@ -0,0 +1,163 @@
+package versioned.host.exp.exponent.modules.universal;
+
+import android.content.Context;
+import android.util.Base64;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import host.exp.exponent.kernel.ExperienceId;
+
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+
+import org.unimodules.core.ModuleRegistry;
+
+import expo.modules.firebase.core.FirebaseCoreService;
+import expo.modules.firebase.core.FirebaseCoreOptions;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import java.io.UnsupportedEncodingException;
+
+public class ScopedFirebaseCoreService extends FirebaseCoreService {
+
+  private ModuleRegistry mModuleRegistry;
+  private String mProtectedAppName;
+  private String mAppName;
+  private FirebaseOptions mAppOptions;
+
+  public ScopedFirebaseCoreService(Context context, JSONObject manifest, ExperienceId experienceId) {
+    super(context);
+
+    // Get the default firebase app name
+    FirebaseApp defaultApp = getFirebaseApp(null);
+    mProtectedAppName = (defaultApp != null) ? defaultApp.getName() : "[DEFAULT]";
+
+    // Get experience key & unique app name
+    String experienceKey = getEncodedExperienceId(experienceId);
+    mAppName = "__sandbox_" + experienceKey;
```

Comment:
Wouldn't `experienceKey` work as well? 🤔

Reply:
Maybe. The core reason I did this was to be coherent with the iOS implementation. On iOS using the `experienceId` failed, because it could contain '%' and '/' characters. The Android implementation seems less restrictive, but I'd rather be on the safe side here and not have any "special" characters in the name, especially if some firebase service may use it for persistent storage.